exchange/exchange.go

```diff
@@ -113,6 +115,18 @@ func (e *Exchange) handleQuery(ctx context.Context, p peer.ID, r Region, q deal.
 	if err != nil {
 		sel = selectors.All()
 	}
+
+	bal, err := e.Wallet().Balance(ctx, e.Wallet().DefaultAddress())
```

Comment:
I'm not sure about this call as it would considerably slow down the ability of this provider to respond to queries (adding up to 600ms depending on lotus node speed). Way to do it would be to subscribe to latest blocks so we can keep watching if there's any changes to the provider balance without fetching it each time we get a retrieval. I have this TODO in my tickets. I would remove this call for now and we can add it later on.

Reply:
But you said : 

> On the provider side we should be checking if there are **sufficient funds in the wallet** as well in order to pay for gas fees when collecting funds. This should be executed at the **exchange.handleQuery** level. I think 0.1FIL should be enough for most gas transactions.

How can I check the balance then?

Reply:
So until you find a way to get the balance without fetching each time the provider we can't do anything on the provider side right? @tchardin

Reply:
Right, let's not do anything for now